baseline/client/js/jspsych-percent-sum.js

```diff
@@ -0,0 +1,91 @@
+jsPsych.plugins["percent-sum"] = (() => {
+    const plugin = {};
+
+    plugin.info = {
+        name: "percent-sum",
+        parameters: {
+            preamble: {
+                type: jsPsych.plugins.parameterType.HTML_STRING,
+                default: undefined,
+            },
+            fields: {
+                type: jsPsych.plugins.parameterType.STRING,
+                array: true,
+                default: undefined,
+            },
+            button_label: {
+                type: jsPsych.plugins.parameterType.STRING,
+                default: "Continue",
+            },
+        },
+    };
+
+    plugin.trial = (display_element, trial) => {
+        // throw if fields are malformed
+        if (trial.fields.length < 1) {
+            throw new Error("fields must not be empty!");
+        } else if ((new Set(trial.fields)).size !== trial.fields.length) {
+            throw new Error("fields must not contain duplicates!");
+        }
+        // build and show display HTML
+        {
+            let html = "";
+            html += `<div id="jspsych-percent-sum-preamble">${trial.preamble}</div>`;
+            html += `<form id="jspsych-percent-sum-form">`;
+            trial.fields.forEach(field => {
+                html += `<label>${field}:`;
+                html += `<input class="jspsych-percent-sum-field" type="number" name="${field}" min="0" max="100" value="0">`;
+                html += `</label>`;
+            });
+            html += `<div id="jspsych-percent-sum-counter-wrapper">`
+            html += `Total: <span id="jspsych-percent-sum-counter">0</span>%`
+            html += `</div>`;
+            html += `<input type="submit" id="jspsych-percent-sum-button" class="jspsych-btn" value="${trial.button_label}" disabled>`;
+            html += `<form/>`;
+            display_element.innerHTML = html;
+        }
+        // find important elements
+        const form = display_element.querySelector("#jspsych-percent-sum-form");
+        const inputs = display_element.querySelectorAll(".jspsych-percent-sum-field");
+        const button = display_element.querySelector("#jspsych-percent-sum-button");
+        const counter = display_element.querySelector("#jspsych-percent-sum-counter");
+        // define validation helper for event listeners
+        const percent_sum = () => {
+            return Array.from(inputs).reduce((sum, inp) => sum + parseInt(inp.value, 10), 0);
+        };
+        // add input listeners
+        inputs.forEach(inp => {
```

Comment:
I missed this the first time around, but jfyi it's often better (performance-wise) to attach a single event handler higher in the hierarchy (in this case on the form, for example) rather than attaching the same handler to multiple different items. The input event will bubble up to the form even though it originates at the input element. No need to change it here, though, as it's (hopefully) unlikely that this plugin will ever have enough fields for it to make a difference.

Reply:
That makes sense, I'll keep that in mind!